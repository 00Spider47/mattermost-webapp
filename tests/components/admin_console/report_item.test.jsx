// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import ReportItem from 'components/admin_console/compliance_reports/report_item/report_item.jsx';

describe('components/ReportItem', () => {
    const baseProps = {
        reports: {
            id: 'someid',
            user_id: 'userid',
            create_at: 123,
            start_at: 1234,
            end_at: 12345,
            emails: [],
            keywords: [],
            type: 'sometype',
            count: 10,
        },
    };

    test('should match snapshot with email', () => {
        const props = {...baseProps, email: 'someemail@somewhere.com'};
        const wrapper = shallow(
            <ReportItem {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with null email', () => {
        const wrapper = shallow(
            <ReportItem {...baseProps}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
