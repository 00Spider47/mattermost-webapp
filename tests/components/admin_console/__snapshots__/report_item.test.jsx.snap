// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ReportItem should match snapshot with email 1`] = `<div />`;

exports[`components/ReportItem should match snapshot with null email 1`] = `<div />`;
